### ISSUE TITLE
Feature request: Pug-like Capture and Assign

### ISSUE THREAD

**Issue by Elbone** [reshape/expressions#27]
In some precompiled template langs we can capture strings and assign them to vars. I'd like something like that.

Perhaps:
```jade
// foo = baz
set(name="my_var" value="{{foo}} bar")
span {{my_var}}
```
renders: 
```html
<span>baz bar</span>
```
and:
```jade
capture(name="my_capture")
  | String with {{foo}}
  | and so on
p {{my_capture}}
```
renders:
```html
<p>String with baz and so on</p>
```
Thoughts?

**Comment by jescalan** [comment]
Yeah, so pug refers to this as "unbuffered code", more info in their docs here: https://pugjs.org/language/code.html

Generally, I see this as an antipattern. Your views are generally not the place to be crunching through logic and code. Because of this, I didn't add the feature, and indeed I have never had a time where I actually needed to use it, instead of doing my processing in code outside the views, before piping the variables in. However, you are certainly not the first person to request this feature, and I have had one person walk me through a situation once where they did legitimately need it. So while I wouldn't be exactly thrilled about adding it, because I feel like it has a GREAT potential for abuse, and enabling people to make messy and unmaintainable codebases, I would be willing to add the feature if someone really did it right.

I think in this case, the best approach would be to follow suit with the way that pug handles it, by adding another delimiter variant that processes code slightly differently. Do you feel like you might be up for this task? I don't currently have the bandwidth or need to add it myself, but if someone were willing to dive in with a PR, I'd be happy to support.
